shell/browser/native_window_mac.mm

```diff
@@ -487,6 +487,7 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   base::scoped_nsobject<NSWindow> child_window(window_,
                                                base::scoped_policy::RETAIN);
   [window_ close];
+  window_ = nil;
```

Comment:
Shouldn't we do this as a result of `windowWillClose` as `CloseImmediately` doesn't take into account the OS closing the window.  This only covers us closing the window programatically.

Reply:
does `windowWillClose` get called as a result of calling `[window_ close]`?

Reply:
I believe it does.

Reply:
yes - it happens regardless of close vs performClose (what changes between the two is whether windowShouldClose is called)